features/unit/v2algodclient.feature

```diff
@@ -0,0 +1,76 @@
+Feature: Algod REST API v2
+
+  Scenario Template: Shutdown
+    When we make any Shutdown call, return mock response "<jsonfile>" and expect error string to contain "<err>"
+    Examples:
+      |jsonfile|err|
+      |TODO    |nil|
+
+  Scenario Template: Register Participation Keys
+    When we make any Register Participation Keys call, return mock response "<jsonfile>" and expect error string to contain "<err>"
+    Examples:
+      |jsonfile|err|
+      |TODO    |nil|
+
+  Scenario Template: Pending Transaction Information
+    When we make any Pending Transaction Information call, return mock response "<jsonfile>" and expect error string to contain "<err>"
+    Then the parsed Pending Transaction Information response should have sender "<sender>"
+    Examples:
+      |jsonfile|err|sender|
+      |TODO    |nil|TODO  |
+
+  Scenario Template: Send Raw Transaction
+    When we make any Send Raw Transaction call, return mock response "<jsonfile>" and expect error string to contain "<err>"
+    Then the parsed Send Raw Transaction response should have txid "<txid>"
+    Examples:
+      |jsonfile|err|txid|
+      |TODO    |nil|TODO|
+
+  Scenario Template: Pending Transactions By Address
+    When we make any Pending Transactions By Address call, return mock response "<jsonfile>" and expect error string to contain "<err>"
+    Then the parsed Pending Transactions By Address response should contain an array of len <len> and element number <idx> should have sender "<sender>"
+    Examples:
+      |jsonfile|err|len|idx|sender|
+      |TODO    |nil|0  |0  |TODO  |
```

Comment:
Could something like this work?
```
  Scenario Template: Pending Transactions By Address
    Given mock http responses in "<jsonfiles>" loaded from "<directory>"
    When I call PendingTransactionsByAddress
    Then the golden is "<golden>" or "<err>"
    Examples:
      | directory          | jsonfiles                                       | err | golden        |
      | responses/algodv2/ | pending_tx_by_address.response,another.response | nil | sdakjhsadg==  |
```

Reply:
Maybe something like this would make it easier to compare results instead of a "golden" hash which isn't human readable.
```
  Scenario Template: Pending Transactions By Address
    Given mock http responses in "<jsonfiles>" loaded from "<directory>"
    When I call PendingTransactionsByAddress
    Then the golden is "<golden>" or "<err>"
    Examples:
      | directory          | jsonfiles                                       | err | golden        |
      | responses/algodv2/ | pending_tx_by_address.response,another.response | nil | addr1,addr2,addr3 |
```